tests/monotouch-test/Vision/VNRequestTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿//
+// Unit tests for VNRequestTests
+//
+// Authors:
+//	Alex Soto <alexsoto@microsoft.com>
+//	
+//
+// Copyright 2018 Microsoft Corporation.
+//
+
+#if !__WATCHOS__ && XAMCORE_2_0
+
+using System;
+using NUnit.Framework;
+
+using Foundation;
+using Vision;
+using System.Collections.Generic;
+
+namespace MonoTouchFixtures.Vision {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class VNRequestTests {
+
+		[SetUp]
+		public void Setup ()
+		{
+			TestRuntime.AssertXcodeVersion (10, 0);
+		}
+
+		[Test]
+		public void VNSupportedRevisionsTest ()
+		{
+			// As of iOS 12, the following classes supports only
+			// revision1 using >= 1 so this does not break in the future
+			// the only intention of this test is to excercise manual code.
+			var v1requests = new List<VNRequestRevision []> {
+				VNCoreMLRequest.SupportedRevisions,
+				VNDetectBarcodesRequest.SupportedRevisions,
+				VNDetectFaceLandmarksRequest.SupportedRevisions,
+				VNDetectFaceRectanglesRequest.SupportedRevisions,
+				VNDetectHorizonRequest.SupportedRevisions,
+				VNDetectRectanglesRequest.SupportedRevisions,
+				VNDetectTextRectanglesRequest.SupportedRevisions,
+				VNTranslationalImageRegistrationRequest.SupportedRevisions,
+				VNHomographicImageRegistrationRequest.SupportedRevisions,
+				VNTrackObjectRequest.SupportedRevisions,
+				VNTrackRectangleRequest.SupportedRevisions,
+			};
+
+			for (int i = 0; i < v1requests.Count; i++)
+				Assert.That (v1requests [i].Length, Is.GreaterThanOrEqualTo (1), $"v1requests[{i}]");
+
+			// As of iOS 12, the following classes supports two revisions
+			var v2requests = new List<VNRequestRevision []> {
+				VNDetectFaceLandmarksRequest.SupportedRevisions,
+				VNDetectFaceRectanglesRequest.SupportedRevisions,
+			};
+
+			for (int i = 0; i < v2requests.Count; i++)
+				Assert.That (v2requests [i].Length, Is.GreaterThanOrEqualTo (2), $"v2requests[{i}]");
+		}
+	}
```

Comment:
more tests are needed to see

* is `0` (Unspecified) ever returned as something supported ?
* what happens when `0` (Unspecified) is used as an **Input** (even if it's part of the **output**) ?
* what happen if `2` is used for an API that supports only `1` ?

Reply:
I disagree, it's just like providing named custom delegates to avoid finding/reading docs while coding.

There's not point in introducing more metadata (e.g. enums over ints) if they don't help the developer make a better decision, i.e. if a revision `X` does not exists for API `A` then it should not be seen (or suggested by intellisense) and if it's only available in some OS version then it should be visible.

Reply:
@spouliot so should we go back just to use nuint then? It seems that the API just does not care at all what you give to it. I agree that the enumification now seems pointless...